agent/engine/task_manager.go

```diff
@@ -472,19 +474,32 @@ func (mtask *managedTask) handleResourceStateChange(resChange resourceStateChang
 		return
 	}
 
+	defer mtask.engine.saver.Save()
+	// Set known status regardless of error so the applied status can be cleared. If there is error,
+	// the known status might be set again below (but that won't affect the applied status anymore).
+	// This follows how container state change is handled.
+	res.SetKnownStatus(status)
 	if err == nil {
-		res.SetKnownStatus(status)
-		mtask.engine.saver.Save()
 		return
 	}
-	seelog.Infof("Managed task [%s]: unable to transition resource %s to %s: %v",
-		mtask.Arn, res.GetName(), res.StatusString(status), err)
-	if status == res.SteadyState() {
-		seelog.Errorf("Managed task [%s]: error while creating resource %s, setting the task's desired status to STOPPED",
-			mtask.Arn, res.GetName())
+
+	if status == res.SteadyState() { // Failed to create resource.
+		seelog.Errorf("Managed task [%s]: failed to create task resource [%s]: %v", mtask.Arn, res.GetName(), err)
+		res.SetKnownStatus(currentKnownStatus) // Set status back to None.
+
+		seelog.Infof("Managed task [%s]: marking task desired status to STOPPED", mtask.Arn)
 		mtask.SetDesiredStatus(apitaskstatus.TaskStopped)
 		mtask.Task.SetTerminalReason(res.GetTerminalReason())
-		mtask.engine.saver.Save()
+	} else { //  Failed to clean up resource.
+		// When failing to clean up a resource, print an error log and assume it's removed, so its dependency can
+		// move forward (e.g. pause container depends on efs volume resource to be removed before moving to stopped).
+		// This doesn't seem to be an ideal way, but seems like we don't deal with failure in task resource cleanup
+		// ever since task resource is added (refer to deleteTask in docker_task_engine.go), so it is still the case here.
+		// Another note: we don't need to mark task desired status to stopped here because we will only try to
+		// clean up the resource when we try to stop the task (i.e. task desired status is already STOPPED).
+		seelog.Warnf("Managed task [%s]: failed to clean up task resource [%s]: %v", mtask.Arn, res.GetName(), err)
```

Comment:
i dont think we need both log lines since they are communicating the same information. Just keep it at warn

Reply:
updated